enterprise/coderd/workspaceproxy.go

```diff
@@ -554,39 +640,105 @@ func (api *API) workspaceProxyRegister(rw http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	// Publish a replicasync event with a nil ID so every replica (yes, even the
+	// current replica) will refresh its replicas list.
+	err = api.Pubsub.Publish(replicasync.PubsubEvent, []byte(uuid.Nil.String()))
+	if err != nil {
+		httpapi.InternalServerError(rw, err)
+		return
+	}
+
+	// Find sibling regions to respond with for derpmesh.
+	siblings := api.replicaManager.InRegion(regionID)
```

Comment:
Is this so the proxy can return these replicas in it's derpmap right after it's registered?

Reply:
This is so the proxy replica that's registering can be aware of all other replicas for that same proxy for derpmesh purposes. Derpmesh allows packets to flow between agent and client even if they're connected to different replicas in the same proxy region by having the replicas forward packets to each other over private networking